src/outputgeneration/ParseTreeWriter.js

```diff
@@ -1565,6 +1565,10 @@ export class ParseTreeWriter extends ParseTreeVisitor {
     this.lastCode_ = value.charCodeAt(value.length - 1);
   }
 
+  writeToken_(token) {
+    this.write_(token.toString());
```

Comment:
So the parameters to `write_()`  now should alway be `string`?  And thus we don't need conversion in `this.writeRaw_(String(value));` ?